contrib/mesos/pkg/scheduler/podtask/pod_task.go

```diff
@@ -51,32 +52,40 @@ const (
 	Deleted  = FlagType("deleted")
 )
 
+var defaultRoles = []string{"*"}
+
 // A struct that describes a pod task.
 type T struct {
 	ID          string
 	Pod         api.Pod
-	Spec        Spec
+	Spec        *Spec
```

Comment:
OK, as does `t.Spec == Spec{}`, if `T.Spec` remained a non-ptr type. If we don't want that littered throughout the code, we can write a `T.HasSpec() bool` func to wrap it.

I'm wary of using ptr's for members that are not read-only in this struct. It's probably due to the mental scars I've accrued from dealing with the race conditions that were littered throughout the original code base.